src/app/views/PublishesConnections.svelte

```diff
@@ -0,0 +1,144 @@
+<script lang="ts">
+  import {getRelayQuality, relaysByUrl} from "@welshman/app"
+  import {ctx} from "@welshman/lib"
+  import {AuthStatus, SocketStatus, type Connection} from "@welshman/net"
+  import {displayRelayUrl} from "@welshman/util"
+  import {quantify} from "hurdak"
+  import {onMount} from "svelte"
+  import AltColor from "src/partials/AltColor.svelte"
+  import SelectButton from "src/partials/SelectButton.svelte"
+
+  export let selected: string
+  export let activeTab: string
+
+  let selectedOptions: string[] = []
+  let connectionsStatus: {[key: string]: Map<string, Connection>} = {}
+
+  const options = [
+    "Connected",
+    "Logging in",
+    "Failed to log in",
+    "Failed to connect",
+    "Waiting to reconnect",
+    "Not connected",
+    "Unstable connection",
+  ]
+
+  const pendingStatuses = [
+    AuthStatus.Requested,
+    AuthStatus.PendingSignature,
+    AuthStatus.PendingResponse,
+  ]
+
+  const failureStatuses = [AuthStatus.DeniedSignature, AuthStatus.Forbidden]
+
+  $: connections = Array.from(ctx.net.pool.data.entries())
+    .filter(([url, cxn]) =>
+      selectedOptions.length ? selectedOptions.some(s => connectionsStatus[s]?.has(url)) : true,
+    )
+    .map(([url, cxn]) => cxn)
+
+  onMount(() => {
+    const interval = setInterval(() => {
+      // make a copy of the connections
+      const newConnectionStatus: {[key: string]: Map<string, Connection>} = {}
+      for (const [url, cxn] of ctx.net.pool.data.entries()) {
+        if (pendingStatuses.includes(cxn.auth.status)) {
+          newConnectionStatus["Logging in"] = (newConnectionStatus["Logging in"] || new Map()).set(
+            url,
+            cxn,
+          )
+        } else if (failureStatuses.includes(cxn.auth.status)) {
+          newConnectionStatus["Failed to log in"] = (
+            newConnectionStatus["Failed to log in"] || new Map()
+          ).set(url, cxn)
+        } else if (cxn.socket.status === SocketStatus.Error) {
+          newConnectionStatus["Failed to connect"] = (
+            newConnectionStatus["Failed to connect"] || new Map()
+          ).set(url, cxn)
+        } else if (cxn.socket.status === SocketStatus.Closed) {
+          newConnectionStatus["Waiting to reconnect"] = (
+            newConnectionStatus["Waiting to reconnect"] || new Map()
+          ).set(url, cxn)
+        } else if (cxn.socket.status === SocketStatus.New) {
+          newConnectionStatus["Not connected"] = (
+            newConnectionStatus["Not connected"] || new Map()
+          ).set(url, cxn)
+        } else if (getRelayQuality(cxn.url) < 0.5) {
+          newConnectionStatus["Unstable connection"] = (
+            newConnectionStatus["Unstable connection"] || new Map()
+          ).set(url, cxn)
+        } else {
+          newConnectionStatus["Connected"] = (newConnectionStatus["Connected"] || new Map()).set(
+            url,
+            cxn,
+          )
+        }
+      }
+      connectionsStatus = newConnectionStatus
+    }, 800)
```

Comment:
This results in a flash of zeroes on first load. Instead, put this in a throttled function and call it on mount and in an interval.

Reply:
it is not updated inside the loop, the connectionStatus is updated once every 800 ms, there is no flashing on each connection state

Reply:
On first page load there is, because the interval doesn't run at first:
https://github.com/user-attachments/assets/2b47224e-536d-4af5-a663-510c9bab2546